Sync/RustSyncManagerAPI.swift

```diff
@@ -0,0 +1,81 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Logger
+import Shared
+@_exported import MozillaAppServices
+
+open class RustSyncManagerAPI {
+    private let logger: Logger
+    let api: SyncManager
```

Comment:
will you be using this directly? wondering if we should make it private 🤔

Reply:
This will be used in the new `RustSyncManager` class (which will be added in the next PR) and exists in the `Providers` directory and will be the replacement to the existing `BrowserSyncManager` class.